### PR TITLE
Fix crashes when `enabledBreadcrumbTypes` is null

### PR DESCRIPTION
**Pull request by imjoehaines**
## Goal

Following on from https://github.com/bugsnag/bugsnag-js/pull/1466, this PR fixes crashes in various plugins that assumed `enabledBreadcrumbTypes` must always be an array. They now allow for the case where it may be null and handle it accordingly

## Testing

Existing tests & new unit tests. Manually tested in browser, ~~electron~~ and ~~react native~~ projects

### CODE REVIEW

File: test/react-native/features/fixtures/ios-module/BugsnagModule.m [hunk 1]
```diff
@@ -62,7 +62,11 @@ @implementation BugsnagModule
     [config setEnabledReleaseStages:[NSSet setWithArray:options[@"enabledReleaseStages"]]];
   }
   if (options[@"enabledBreadcrumbTypes"] != nil) {
-    [config setEnabledBreadcrumbTypes:[NSSet setWithArray:options[@"enabledBreadcrumbTypes"]]];
+    if ([options[@"enabledBreadcrumbTypes"] isEqual:[NSNull null]]) {
```
Comment (kattrali) [inline test/react-native/features/fixtures/ios-module/BugsnagModule.m:65]:
```suggestion
if (options[@"enabledBreadcrumbTypes"] && ![options[@"enabledBreadcrumbTypes"] isEqual:[NSNull null]]) {
```

**Review by kattrali — APPROVED** [review]
Is it possible to either:

* Coerce `enabledBreadcrumbTypes` to always be an array during config validation (where null -> `all`)? or
* Add a helper on config to handle the null check? Most of the validation checks now are of the form `if (config.breadcrumbTypes !== null && config.breadcrumbTypes.contain(type))` where maybe it could be something like `if (config.isBreadcrumbTypeEnabled(type))`

Trying to think of a way to prevent this kind of issue going forward since every future plugin/consumer of this config option basically needs to have this check. Stronger type validation on our part? Additional linting?

File: test/react-native/features/fixtures/ios-module/BugsnagModule.m (hunk 1 above)
Comment (kattrali) [reply test/react-native/features/fixtures/ios-module/BugsnagModule.m:65]: I meant to leave it as the default, treating `nil` and `NSNull` the same

**Comment by imjoehaines** [comment]
> Trying to think of a way to prevent this kind of issue going forward since every future plugin/consumer of this config option basically needs to have this check. Stronger type validation on our part? Additional linting?

I avoided doing anything extra in core to try to avoid increasing the browser bundle size, but it turns out that a helper method actually saves space because it removes duplication from plugins

I've opened #1468 for this as it affects files in this PR, #1466 and files that weren't touched by either PR

**Pull request merged**